drf_registration/api/login.py

```diff
@@ -132,12 +133,20 @@ def post(self, request, *args, **kwargs):
         try:
             user = User.objects.get(email=user_data['email'])
         except User.DoesNotExist:
-            user = User.objects.create(
-                username=user_data['email'],
-                email=user_data['email'],
-                first_name=user_data.get('first_name'),
-                last_name=user_data.get('last_name'),
-            )
+            if (provider == FACEBOOK_PROVIDER):
```

Comment:
@DinithHerath Many thanks for your update. I think we don't need to check the provider, the implementation like this:
```
 user = User.objects.create(
    username=user_data['email'],
    email=user_data['email'],
    first_name=user_data.get('first_name') or user_data.get('given_name'),
    last_name=user_data.get('last_name') or user_data.get('family_name'),
 )
```
Feel free to check and update, I will merge after that. Sorry for late response.